### PR TITLE
cody: migrate site-admin>embeddings to FilteredConnection

### PR DESCRIPTION
**Pull request by stefanhengl**
This is a refactor with no UI changes.

Why?

We want to use the `FilteredConnection` component (just like code intel does for their precise index list) because it offers "search" and "filters" which make it much nicer to handle hundreds or thousands of embedding jobs.

This is just the first step. In the next PRs I will add "search" and "filter" and probably update the design. 

This PR:
Screenshot 2023-06-01 at 12 35 17

## Test plan
- `sg start embeddings`
- Create new embedding jobs
- Compare to existing UI/UX

### CODE REVIEW

File: client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx [hunk 1]
```diff
@@ -130,37 +131,35 @@ export const SiteAdminCodyPage: FC<SiteAdminCodyPageProps> = ({ telemetryService
...
-                                noun="repository embedding job"
-                                pluralNoun="repository embedding jobs"
-                                hasNextPage={hasNextPage}
-                                emptyElement={<EmptyList />}
-                            />
-                            {hasNextPage && <ShowMoreButton centered={true} onClick={fetchMore} />}
-                        </SummaryContainer>
-                    )}
-                </ConnectionContainer>
+                <FilteredConnection<
```
Comment (mrnugget) [inline client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx:134]: FilteredConnection is actually "deprecated" and what you had before is the recommended way. There's also examples of using search with the `ConnectionList`: https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/team/list/TeamListPage.tsx?L119-123

I understand though that this is easier, but maybe someone from #dev-frontend can help (at least make informed decision on what the best approach is here)?

File: client/web/src/enterprise/site-admin/cody/backend.ts [hunk 2]
```diff
@@ -35,38 +34,46 @@ const REPO_EMBEDDING_JOB_FRAGMENT = gql`
     }
 `
 
-export const REPO_EMBEDDING_JOBS_LIST_QUERY = gql`
+const REPO_EMBEDDING_JOB_CONNECTION_FIELDS_FRAGMENT = gql`
     ${REPO_EMBEDDING_JOB_FRAGMENT}
+    fragment RepoEmbeddingJobConnectionFields on RepoEmbeddingJobsConnection {
+        totalCount
+        pageInfo {
```
Comment (mrnugget) [inline client/web/src/enterprise/site-admin/cody/backend.ts:41]: Have you seen this? https://docs.sourcegraph.com/dev/background-information/web/graphql#best-practices-for-paginating-resources-with-graphql

File: client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx (hunk 1 above)
Comment (stefanhengl) [reply client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx:134]: FilteredConnection is not marked as deprecated. How could I have found this out before spending a full day on this? Reaching out to #dev-frontend.

File: client/web/src/enterprise/site-admin/cody/backend.ts (hunk 2 above)
Comment (stefanhengl) [reply client/web/src/enterprise/site-admin/cody/backend.ts:41]: I didn't touch the logic, I just needed to have a fragment for the autogenerated type. 

I am guessing you would want to return "startCursor" as well?

File: client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx (hunk 1 above)
Comment (mrnugget) [reply client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx:134]: > How could I have found this out before spending a full day on this?

I hear you. It's tricky. I have had to do the same dance last year: https://github.com/sourcegraph/sourcegraph/pull/39134 There's no docs, no canonical examples. It's mostly "the newer pages use this new thing, make sure to look at newer pages". (Kinda like in our backend and how we do some database/store stuff 😬 )

Comment (valerybugakov) [reply client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx:134]: > FilteredConnection is not marked as deprecated. 

That's fair, and it's a mistake from our side. We've been working on replacing `FilteredConnection` for a long time and should've marked it as deprecated. 

> We want to use the FilteredConnection component (just like code intel does for their precise index list) because it offers "search" and "filters"

The `FilteredConnection` is deprecated in favor of smaller composable components. Here's [an example](https://github.com/sourcegraph/sourcegraph/blob/4794d2ff1669a83bb15aa4e2ee8c448e53eae754/client/web/src/team/list/TeamListPage.tsx#L106-L147) of how to implement search and filters using underlying primitives. Do you think it looks helpful? 

> There's no docs, no canonical examples.

We should add a docstring to the `FilteredConnection` component and point to some existing examples in the codebase. Ideally, yeah, we need docs.

In summary, the recommended approach is to use `FilteredConnection` primitives instead of the monolithic component. cc @umpox, who worked on the initial component decomposition, and @philipp-spiess, who lead the UI strategic readiness effort where we reviewed our pagination patterns (please chime in if I missed anything!).

Comment (valerybugakov) [reply client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx:134]: Here's the minimal thing we should do if my understandinging above is not completely out-of-date:
- https://github.com/sourcegraph/sourcegraph/pull/52823

Comment (stefanhengl) [reply client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx:134]: Thanks for the comments and all the pointers! I will close this PR.

Comment (mrnugget) [reply client/web/src/enterprise/site-admin/cody/SiteAdminCodyPage.tsx:134]: @valerybugakov @umpox it would be great if we had at least a bare-bones documentation page (or a storybook story!) that shows a component using `useConnection` and `ConnectionContainer` etc. If this example here https://github.com/sourcegraph/sourcegraph/pull/52823/files#r1214090420 for example would be on a separate Markdown docs page we could link to, that'd be great
